lock-api/src/main/java/com/palantir/lock/client/LeasingTimelockClient.java

```diff
@@ -0,0 +1,160 @@
+        return delegate.getFreshTimestamps(numTimestampsRequested);
+    }
+
+    @Override
+    public LockImmutableTimestampResponse lockImmutableTimestamp(IdentifiedTimeLockRequest request) {
+        return delegate.lockImmutableTimestamp(request);
+    }
+
+    @Override
+    public StartAtlasDbTransactionResponse startAtlasDbTransaction(IdentifiedTimeLockRequest request) {
+        return delegate.startAtlasDbTransaction(request);
+    }
+
+    @Override
+    public StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransaction(
+            StartIdentifiedAtlasDbTransactionRequest request) {
+        LeasableStartIdentifiedAtlasDbTransactionResponse leasableResponse =
+                delegate.leasableStartIdentifiedAtlasDbTransaction(request);
+
+        StartIdentifiedAtlasDbTransactionResponse response = leasableResponse.getStartTransactionResponse();
+        Lease lease = leasableResponse.getLease();
+        LeasedLockToken leasedLockToken = LeasedLockToken.of(response.immutableTimestamp().getLock(), lease);
+        long immutableTs = response.immutableTimestamp().getImmutableTimestamp();
+
+        return ImmutableStartIdentifiedAtlasDbTransactionResponse.of(
+                ImmutableLockImmutableTimestampResponse.of(immutableTs, leasedLockToken),
+                response.startTimestampAndPartition());
+    }
+
+    @Override
+    public long getImmutableTimestamp() {
+        return delegate.getImmutableTimestamp();
+    }
+
+    @Override
+    public LockResponse lock(LockRequest request) {
+        LeasableLockResponse leasableResponse = delegate.leasableLock(request);
+
+        LockResponse lockResponse = leasableResponse.getLockResponse();
+        Lease lease = leasableResponse.getLease();
+        LeasedLockToken leasedLockToken = LeasedLockToken.of(lockResponse.getToken(), lease);
+
+        return ImmutableLockResponse.of(Optional.of(leasedLockToken));
+    }
+
+    @Override
+    public WaitForLocksResponse waitForLocks(WaitForLocksRequest request) {
+        return delegate.waitForLocks(request);
+    }
+
+    @Override
+    public Set<LockToken> refreshLockLeases(Set<LockToken> tokens) {
+        IdentifiedTime identifiedTime = delegate.getLeaderTime();
+        Set<LeasedLockToken> allTokens = tokens.stream()
+                .map(token -> (LeasedLockToken) token)
+                .collect(Collectors.toSet());
+
+        Set<LeasedLockToken> validByLease = allTokens.stream()
+                .filter(token -> token.isValid(identifiedTime))
+                .collect(Collectors.toSet());
+
+        Set<LeasedLockToken> toRefresh = Sets.difference(allTokens, validByLease);
+
+        Set<LeasedLockToken> refreshedTokens = ImmutableSet.of();
+
+        if (!toRefresh.isEmpty()) {
+            LeasableRefreshLockResponse refreshLockResponse = delegate.leasableRefreshLockLeases(
+                    toRefresh.stream().map(LeasedLockToken::serverToken).collect(Collectors.toSet()));
```

Comment:
this snippet could again be a method `serverTokens(Set<LeasedLockToken> leasedTokens)`